Sources/SwiftFusion/Inference/PenguinExtensions.swift

```diff
@@ -104,6 +137,577 @@ extension ArrayBuffer: CustomDebugStringConvertible {
+  
+  let slowPathCache: VTableCache
+  
+  if let cache = TLS.get(vTableCacheKey) {
+    if let r = cache.tables[tableID] { return r.assumingMemoryBound(to: Table.self) }
+    slowPathCache = cache
+  }
+  else {
+    slowPathCache = makeVTableCache()
+  }
+  return makeAndCacheTable(in: slowPathCache)
+}
+
+// =================================================================================================
+
+/// “Existential VTable” for requirements that can't be threaded through closures (see
+/// https://forums.swift.org/t/pitch-rethrows-unchecked/10078/9).
+fileprivate protocol AnyMutableCollectionExistentialDispatchProtocol {
+  static func withContiguousStorageIfAvailable(
+    _ self_: AnyValue,
+    _ body: (UnsafeRawPointer?, Int) throws -> Void
+  ) rethrows
+
+  static func withUnsafeMutableBufferPointerIfSupported(
+    _ self_: inout AnyValue,
+    _ body: (UnsafeMutableRawPointer?, Int) throws -> Void
+  ) rethrows
+
+  static func withContiguousMutableStorageIfAvailable(
+    _ self_: inout AnyValue,
+    _ body: (UnsafeMutableRawPointer?, Int) throws -> Void
+  ) rethrows
+
+  static func partition(
+    _ self_: inout AnyValue,
+    _ resultIndex: UnsafeMutableRawPointer,
+    _ belongsInSecondPartition: (UnsafeRawPointer) throws -> Bool
+  ) rethrows
+}
+
+/// Instance of `AnyMutableCollectionExistentialDispatchProtocol` for an AnyMutableCollection
+/// wrapping `C`.
+enum AnyMutableCollectionExistentialDispatch<C: MutableCollection>
+  : AnyMutableCollectionExistentialDispatchProtocol {
+  static func withContiguousStorageIfAvailable(
+    _ self_: AnyValue,
+    _ body: (UnsafeRawPointer?, Int) throws -> Void
+  ) rethrows {
+    try self_[unsafelyAssuming: Type<C>()]
+      .withContiguousStorageIfAvailable { try body($0.baseAddress, $0.count) }
+  }
+
+  static func withUnsafeMutableBufferPointerIfSupported(
+    _ self_: inout AnyValue,
+    _ body: (UnsafeMutableRawPointer?, Int) throws -> Void
+  ) rethrows {
+    try self_[unsafelyAssuming: Type<C>()]
+      ._withUnsafeMutableBufferPointerIfSupported { try body($0.baseAddress, $0.count) }
+  }
+
+  static func withContiguousMutableStorageIfAvailable(
+    _ self_: inout AnyValue,
+    _ body: (UnsafeMutableRawPointer?, Int) throws -> Void
+  ) rethrows {
+    try self_[unsafelyAssuming: Type<C>()]
+      .withContiguousMutableStorageIfAvailable { try body($0.baseAddress, $0.count) }
+  }
+
+  static func partition(
+    _ self_: inout AnyValue,
+    _ resultIndex: UnsafeMutableRawPointer,
+    _ belongsInSecondPartition: (UnsafeRawPointer) throws -> Bool
+  ) rethrows {
+    let r = try self_[unsafelyAssuming: Type<C>()]
+      .partition { try withUnsafePointer(to: $0) { try belongsInSecondPartition($0) } }
+    resultIndex.assumingMemoryBound(to: AnyMutableCollection<C.Element>.Index?.self)
+      .pointee = .init(r)
+  }
+}
+
+/// A type-erased mutable collection similar to the standard library's `AnyCollection`, but with
+/// parameterized type-erasure substrate.
```

Comment:
Honestly, I have no idea what I meant by that!
```suggestion
/// A type-erased mutable collection similar to the standard library's `AnyCollection`.
```